packages/apps/staking/src/components/modals/KVStoreModal.tsx

```diff
@@ -118,9 +130,16 @@ const KVStoreModal: React.FC<Props> = ({
   };
 
   const handleDelete = (index: number) => {
-    const updatedData = formData.filter((_, i) => i !== index);
-    updatePendingChanges(formData[index].key, '');
-    setFormData(updatedData);
+    setItemToDeleteIndex(index);
+  };
+
+  const handleConfirmDelete = () => {
```

Comment:
actually we delete nothing here, as if you close the modal and open it again, then the kv pair would get back. prolly need to talk to design